src/build/cargo.rs

```diff
@@ -299,20 +308,29 @@ impl Executor for RlsExecutor {
             }
         }
 
-        if self.workspace_mode {
-            let output = cmd.output().expect("Couldn't execute rustc");
-            let mut stderr_json_msg = convert_message_to_json_strings(output.stderr);
-            self.compiler_messages.lock().unwrap().append(&mut stderr_json_msg);
-        } else {
-            cmd.status().expect("Couldn't execute rustc");
-        }
-
         // Finally, store the modified cargo-generated args/envs for future rustc calls
         args.insert(0, rustc_exe);
         let mut compilation_cx = self.compilation_cx.lock().unwrap();
         compilation_cx.args = args;
         compilation_cx.envs = cargo_cmd.get_envs().clone();
 
+        if self.workspace_mode {
+            let args = &compilation_cx.args;
+            let envs = &compilation_cx.envs;
```

Comment:
Only `cargo` routine populates the args and envs, `rustc` consumes them externally. In workspace mode currently only ever `cargo` routine is [called](https://github.com/Xanewok/rls/blob/master/src/build/mod.rs#L386), which then prepares RLS-specific args/envs from those generated by Cargo itself as usual. Then it feeds them directly to, introduced in this change, `rustc::rustc` (unlike being passed via `Command` API and actually executing the command). In practice it doesn't use use `compilation_cx.{args,envs}` at all, however it doesn't prevent it from being set [here](https://github.com/rust-lang-nursery/rls/pull/424/files/85e20c0b8ec394bbfcc58203aaeb610dca1fe1e4#diff-7946bdad71ef1f573a8d51890bda997fR314)